app/src/main/java/com/nio/tcp/handler/SelectionKeyEventHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.nio.tcp.handler;
+
+import com.nio.tcp.exception.OperationNotPresentException;
+import com.nio.tcp.handler.helper.EventHelper;
+
+import java.nio.channels.SelectionKey;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SelectionKeyEventHandler implements EventHandler {
+
+    private static final String OPERATION_NOT_PRESENT_MESSAGE = "Handler for operation with code %s not present";
+    private static EventHandler selectionKeyEventHandlerInstance;
+    private final Map<Integer, EventHelper> eventHelperMap;
+
+    private SelectionKeyEventHandler() {
+        eventHelperMap = new HashMap<>();
+    }
+
+    public static EventHandler getInstance() {
+        if (selectionKeyEventHandlerInstance == null) {
+            selectionKeyEventHandlerInstance = new SelectionKeyEventHandler();
+        }
+        return selectionKeyEventHandlerInstance;
+    }
+
+    @Override
+    public void initHelper(Integer operationCode, EventHelper eventHelper) {
```

Comment:
As I perceive it from the name of the method, it should be called only once.
But if we take a look to the body of method the sence of the method is populating map, so it can be called only once.